dartsim/src/SDFFeatures_TEST.cc

```diff
@@ -158,20 +158,26 @@ TEST(SDFFeatures_TEST, CheckDartsimData)
     EXPECT_DOUBLE_EQ(0.0, axis[2]);
   }
 
-  const dart::dynamics::SkeletonPtr freeBody =
+  {
+    const dart::dynamics::SkeletonPtr freeBody =
       dartWorld->getSkeleton("free_body");
-  ASSERT_NE(nullptr, freeBody);
-  ASSERT_EQ(1u, freeBody->getNumBodyNodes());
-  const dart::dynamics::BodyNode *bn = freeBody->getBodyNode(0);
-  ASSERT_NE(nullptr, bn);
-
-  EXPECT_TRUE(dynamic_cast<const dart::dynamics::FreeJoint*>(
-                bn->getParentJoint()));
-
-  const Eigen::Vector3d translation = bn->getTransform().translation();
-  EXPECT_DOUBLE_EQ(0.0, translation[0]);
-  EXPECT_DOUBLE_EQ(10.0, translation[1]);
-  EXPECT_DOUBLE_EQ(10.0, translation[2]);
+    ASSERT_NE(nullptr, freeBody);
+    ASSERT_EQ(1u, freeBody->getNumBodyNodes());
+    const dart::dynamics::BodyNode *bn = freeBody->getBodyNode(0);
+    ASSERT_NE(nullptr, bn);
+
+    EXPECT_TRUE(dynamic_cast<const dart::dynamics::FreeJoint*>(
+          bn->getParentJoint()));
+
+    const Eigen::Vector3d translation = bn->getTransform().translation();
+    EXPECT_DOUBLE_EQ(0.0, translation[0]);
+    EXPECT_DOUBLE_EQ(10.0, translation[1]);
+    EXPECT_DOUBLE_EQ(10.0, translation[2]);
+
+    const dart::dynamics::ShapeNode *collision1 = bn->getShapeNode(0);
+    auto aspect = collision1->getDynamicsAspect();
+    EXPECT_DOUBLE_EQ(0.8, aspect->getRestitutionCoeff());
```

Comment:
This test failed on GitHub actions Focal CI:

```
  [ RUN      ] SDFFeatures_TEST.CheckDartsimData
  [Err] [SDFFeatures.cc:814] Asked to construct a joint of sdf::JointType [7], but that is not supported yet. Creating a FIXED joint instead
  [Err] [SDFFeatures.cc:814] Asked to construct a joint of sdf::JointType [4], but that is not supported yet. Creating a FIXED joint instead
  [Err] [SDFFeatures.cc:814] Asked to construct a joint of sdf::JointType [2], but that is not supported yet. Creating a FIXED joint instead
  /github/workspace/dartsim/src/SDFFeatures_TEST.cc:179: Failure
  Expected equality of these values:
    0.8
      Which is: 0.80000000000000004
    aspect->getRestitutionCoeff()
      Which is: 0
  [  FAILED  ] SDFFeatures_TEST.CheckDartsimData (128 ms)
```

Is this related to the DART version?

Reply:
Yes, the coefficient was only being set for DART>=6.10.0. 3e6310a fixes it, but note that for older versions of DART, the restitution coefficient has to be set on the BodyNode instead of the shape. See https://github.com/dartsim/dart/issues/200